### PR TITLE
Create PR Fixes in Temp Directory

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.

---

### CODE REVIEW

File: commands/utils/git.go [hunk 1]
```diff
@@ -41,11 +66,9 @@ func (gm *GitManager) CreateBranchAndCheckout(branchName string) error {
 }
 
 func (gm *GitManager) createBranchAndCheckout(branchName string, create bool) error {
-	// branchName can be short name (master) or full name (refs/heads/master)
-	shortBranchName := plumbing.ReferenceName(branchName).Short()
 	checkoutConfig := &git.CheckoutOptions{
 		Create: create,
-		Branch: plumbing.NewBranchReferenceName(shortBranchName),
+		Branch: getBranch(branchName),
```
Comment (sverdlov93) [inline commands/utils/git.go:71]:
```suggestion
		Branch: getFullBranchName(branchName),
```

File: commands/utils/git.go [hunk 2]
```diff
@@ -146,3 +166,15 @@ func (gm *GitManager) IsClean() (bool, error) {
...
 }
+
+func createBasicAuth(token string) *http.BasicAuth {
+	return &http.BasicAuth{
+		Username: "username", // The username can be anything except an empty string
+		Password: token,
+	}
+}
+
+func getBranch(branchName string) plumbing.ReferenceName {
```
Comment (sverdlov93) [inline commands/utils/git.go:177]:
```suggestion
// Get full branch name ( for example: refs/heads/master)
func getFullBranchName(branchName string) plumbing.ReferenceName {
```

File: commands/createfixpullrequests.go [hunk 3]
```diff
@@ -42,19 +43,55 @@ func (cfp *CreateFixPullRequestsCmd) scan(params *utils.FrogbotParams) ([]servic
...
+		e := restoreDir()
+		if err == nil {
+			err = e
+		}
+	}()
+
+	// Fix all impacted packages
 	for impactedPackage, fixVersionInfo := range fixVersionsMap {
-		clientLog.Info("Fixing", impactedPackage, "with", fixVersionInfo.fixVersion)
+		clientLog.Info("Start fixing", impactedPackage, "with", fixVersionInfo.fixVersion)
```
Comment (sverdlov93) [inline commands/createfixpullrequests.go:94]: maybe add a log.info("------------...)
at the start of each loop

File: commands/utils/git.go [hunk 4]
```diff
@@ -32,6 +34,29 @@ func (gm *GitManager) Checkout(branchName string) error {
 	return err
 }
 
+func (gm *GitManager) Clone(gitToken, destinationPath, branchName string) error {
+	// Gets the remote repo url from current .git dir
```
Comment (eyalbe4) [inline commands/utils/git.go:38]: ...from current .git dir... --> ...from the current .git dir...

File: commands/utils/git.go [hunk 5]
```diff
@@ -32,6 +34,29 @@ func (gm *GitManager) Checkout(branchName string) error {
...
+	gitRemote, err := gm.repository.Remote(gm.remoteName)
+	if err != nil || len(gitRemote.Config().URLs) < 1 {
+		return fmt.Errorf("'git remote %s' failed with error: %s", gm.remoteName, err.Error())
```
Comment (eyalbe4) [inline commands/utils/git.go:41]: We will fail here due to nil pointer, if err is nil.

File: commands/utils/git.go [hunk 6]
```diff
@@ -32,6 +34,29 @@ func (gm *GitManager) Checkout(branchName string) error {
...
+		Auth:          createBasicAuth(gitToken),
+		RemoteName:    gm.remoteName,
+		ReferenceName: getFullBranchName(branchName),
+	}
+	repo, err := git.PlainClone(destinationPath, false, cloneOptions)
+	if err != nil {
+		return fmt.Errorf("'git clone %s from %s' failed with error: %s", branchName, repoURL, err.Error())
+	}
+	gm.repository = repo
+	clientLog.Debug(fmt.Sprintf("project cloned from %s to %s", repoURL, destinationPath))
```
Comment (eyalbe4) [inline commands/utils/git.go:56]: project --> Project

File: commands/utils/git.go [hunk 7]
```diff
@@ -146,3 +166,16 @@ func (gm *GitManager) IsClean() (bool, error) {
 
 	return status.IsClean(), nil
 }
+
+func createBasicAuth(token string) *http.BasicAuth {
+	return &http.BasicAuth{
+		Username: "username", // The username can be anything except an empty string
```
Comment (eyalbe4) [inline commands/utils/git.go:172]: Let's put the comment above the line is descrives.

... except an empty string --> ... except for an empty string

File: commands/utils/git.go [hunk 8]
```diff
@@ -146,3 +166,16 @@ func (gm *GitManager) IsClean() (bool, error) {
...
+		Password: token,
+	}
+}
+
+// getFullBranchName returns the full branch name ( for example: refs/heads/master)
```
Comment (eyalbe4) [inline commands/utils/git.go:177]: ...( for example:... --> ...(for example:...

File: commands/utils/git.go [hunk 9]
```diff
@@ -146,3 +166,16 @@ func (gm *GitManager) IsClean() (bool, error) {
...
+// The input branchName can be short name (master) or full name (refs/heads/master)
```
Comment (eyalbe4) [inline commands/utils/git.go:178]: ...be short name (master) or full name... --> ...be a short name (master) or a full name..

File: commands/createfixpullrequests.go [hunk 10]
```diff
@@ -42,19 +43,56 @@ func (cfp *CreateFixPullRequestsCmd) scan(params *utils.FrogbotParams) ([]servic
...
+	// Clone the content of the repo to the new working directory
 	gitManager, err := utils.NewGitManager(".", "origin")
 	if err != nil {
 		return err
 	}
+	err = gitManager.Clone(params.Token, wd, params.BaseBranch)
+	if err != nil {
+		return err
+	}
+	// Cd to the temp working directory
```
Comment (eyalbe4) [inline commands/createfixpullrequests.go:80]: ...Cd to the temp... --> ...'CD' into the temp...

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**